xarray/core/rolling.py

```diff
@@ -512,8 +622,11 @@ def _bottleneck_reduce(self, func, keep_attrs, **kwargs):
             values = values[valid]
 
         attrs = self.obj.attrs if keep_attrs else {}
+        output_dim_coords = self._get_rolling_dim_coords()
```

Comment:
Yes I think we should avoid the sel in general or at least change it to a slice to prevent an unnecessary copy.

Here's how I tested that out.

``` python
a = xr.DataArray([1, 2, 3], dims="x", coords=dict(x=[0, 1, 2]))
print(a.data.flags["OWNDATA"])
print(a.sel(x=slice(None, None)).data.flags["OWNDATA"])
print(a.sel(a.coords).data.flags["OWNDATA"])
```
```
True
False
True  # this means it's independent of `a`
```

We can also defer this to later, and use the other code path if `pad is False`

Reply:
I think I addressed the other comments.  I'll try to work on this tomorrow.

Reply:
To address the error that @mathause pointed out, I had to change this to an `isel`, and am passing a dictionary of slices.  It still is probably possible to do something more efficient and remove the call to `pad` above in some cases, but it's a little tedious to work out, and I'm not sure how much more time I can spend on this right now.